src/datepicker/datepicker.component.ts

```diff
@@ -190,13 +195,23 @@ export class DatePicker implements OnDestroy, OnChanges, AfterViewChecked {
 		}
 	}
 
+	ngAfterViewInit() {
+		const scrollObservable = scrollableParentsObservable(this.elementRef.nativeElement);
+		this.scrollSubscription = scrollObservable.subscribe((event: any) => {
+			if (!isVisibleInContainer(this.elementRef.nativeElement, event.target)) {
+				this.flatpickrInstance.close();
+			} else {
+				this.flatpickrInstance._positionCalendar();
+			}
```

Comment:
Sorry, one last thing, I think we'll need to check `isFlatpickrLoaded` ... these unchecked calls may error

Reply:
done!